presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPushProjectionThroughJoin.java

```diff
@@ -124,4 +125,23 @@ public void testDoesNotPushStraddlingProjection()
         Optional<PlanNode> rewritten = pushProjectionThroughJoin(planNode, noLookup(), new PlanNodeIdAllocator());
         assertThat(rewritten).isEmpty();
     }
+
+    @Test
+    public void testDoesNotPushProjectionThroughOuterJoin()
```

Comment:
I'd love to see full integration test for this (SQL input) too

Reply:
added test